core/templates/pages/contributor-dashboard-page/modal-templates/translation-suggestion-review-modal.component.ts

```diff
@@ -264,43 +318,106 @@ export class TranslationSuggestionReviewModalComponent implements OnInit {
     const response = await this.threadDataBackendApiService.fetchMessagesAsync(
       threadId);
     const threadMessageBackendDicts = response.messages;
-    this.reviewMessage = threadMessageBackendDicts.map(
-      m => ThreadMessage.createFromBackendDict(m))[1].text;
+    let threadMessages = threadMessageBackendDicts.map(
+      m => ThreadMessage.createFromBackendDict(m));
+    // This is to prevent a console error when a contribution
+    // doesn't have a review message. When a contribution has
+    // a review message the second element of the threadMessages
+    // array contains the actual review message.
+    if (threadMessages[1] !== undefined) {
+      this.reviewMessage = threadMessages[1].text;
+    }
   }
 
-  showNextItemToReview(suggestionId: string): void {
-    this.resolvedSuggestionIds.push(this.activeSuggestionId);
-    if (this.lastSuggestionToReview) {
+  refreshModalData(): void {
+    this.activeSuggestion = this.activeContribution.suggestion;
+    if (this.activeContribution.details === null) {
+      return;
+    }
+    this.activeContributionDetails = this.activeContribution.details;
```

Comment:
hm do we need to update this when we advance next or previous as well? The `this.activeContributionDetails` variable doesn't seem to gain us much, so maybe get rid of it and just call `this.activeContribution.details`.

Reply:
`this.activeContributionDetails` is used in the frontend to render some data (eg. chapter_title)

Reply:
Yeah, but I mean we should just call `activeContribution.details` in the frontend so that we don't need to maintain the `this.activeContributionDetails` variable.